toffy/json_utils_test.py

```diff
@@ -77,3 +78,44 @@ def test_list_moly_fovs(tmpdir):
     pred_moly_fovs = json_utils.list_moly_fovs(tmpdir)
 
     assert np.array_equal(pred_moly_fovs.sort(), moly_fovs.sort())
+
+
+def test_read_json_file():
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+
+        # create fake jsons
+        moly_json = {'name': 'bob',
+                     'standardTarget': 'Molybdenum Foil'}
+        json_path = tmp_dir+"/test.json"
+        
+        # write test json
+        with open(json_path, 'w') as jp:
+            json.dump(moly_json, jp)
+
+        # Make sure errors come up when the directory is bad. Doesnt check to see
+        # if filename itself exists somewhere else bc that would involve recursion into directories
+        # This relies upon io_utils.validate_paths() being fully functional and tested
+        bad_path = "/neasdf1246ljea/asdfje12ua3421ndsf/asdf.json"
+        with pytest.raises(ValueError, match=f'The path, {bad_path}, is not prefixed with \'../data\'.\n'
+                    f'Be sure to add all images/files/data to the \'data\' folder, '
+                    f'and to reference as \'../data/path_to_data/myfile.tif\''):
+                json_utils.read_json_file(bad_path)
+
+        # Create bad dir path to invoke all 3 possibilities of validate_paths()
+        # not sure how to do this because i dont know what the dir structure is supposed to be
+        
+        # Make sure errors raised when dirpath is good but file is bad
+        # I guess dont need this because native python already has "file does not exist" errors?
+        # But also not sure why this is allowed to pass PyTest because read_json_file
+        # should return a file not found error?
+        with pytest.raises(ValueError, match=''):
+               json_utils.read_json_file(tmp_dir+"/dkwn4823hjf08371gjsdfasdfa20135ndjsa.json")
+
+        # Read json with read_json_file function assuming file path is good
+        newfile_test = json_utils.read_json_file(json_path)
+
+        # Make sure using the read_json_file leads to the same object as moly_json
+        assert newfile_test == moly_json
+
+    return
```

Comment:
Now you can add the same tests for `read_json`. One test case where it's valid, one test where it's not.